kDrive/UI/Controller/Files/Save File/SaveFileViewController.swift

```diff
@@ -478,12 +478,16 @@ extension SaveFileViewController: SelectPhotoFormatDelegate {
 // MARK: - FooterButtonDelegate
 
 extension SaveFileViewController: FooterButtonDelegate {
-    @objc func didClickOnButton() {
+    @objc func didClickOnButton(_ sender: AnyObject) {
         guard let selectedDriveFileManager,
               let selectedDirectory else {
             return
         }
 
+        // Making sure the user cannot spam the button on tasks that may take a while
+        let button = sender as? IKLargeButton
+        button?.isEnabled = false
```

Comment:
Maybe we can set a loading state in this case?

Reply:
Good catch, thanks for the `.setLoading` tip.